tests/MongoDB.Driver.Tests/Specifications/retryable-writes/prose-tests/CommandConstructionTests.cs

```diff
@@ -0,0 +1,332 @@
+        {
+            RequireServer.Check().ClusterTypes(ClusterType.ReplicaSet, ClusterType.Sharded);
+
+            DropCollection();
+            var eventCapturer = CreateEventCapturer();
+            using (var client = CreateDisposableClient(eventCapturer))
+            {
+                var database = client.GetDatabase(_databaseName);
+                var collection = database.GetCollection<BsonDocument>(_collectionName);
+
+                switch (operation)
+                {
+                    case "deleteOne":
+                        var deleteOneFilter = Builders<BsonDocument>.Filter.Eq("_id", 1);
+                        collection.DeleteOne(deleteOneFilter);
+                        break;
+
+                    case "findOneAndDelete":
+                        var findOneAndDeleteFilter = Builders<BsonDocument>.Filter.Eq("_id", 1);
+                        collection.FindOneAndDelete(findOneAndDeleteFilter);
+                        break;
+
+                    case "findOneAndReplace":
+                        var findOneAndReplaceFilter = Builders<BsonDocument>.Filter.Eq("_id", 1);
+                        var findOneAndReplaceReplacement = new BsonDocument("_id", 1);
+                        collection.FindOneAndReplace(findOneAndReplaceFilter, findOneAndReplaceReplacement);
+                        break;
+
+                    case "findOneAndUpdate":
+                        var findOneAndUpdateFilter = Builders<BsonDocument>.Filter.Eq("_id", 1);
+                        var findOneAndUpdateUpdate = Builders<BsonDocument>.Update.Set("x", 2);
+                        collection.FindOneAndUpdate(findOneAndUpdateFilter, findOneAndUpdateUpdate);
+                        break;
+
+                    case "insertOne":
+                        var document = new BsonDocument("_id", 1);
+                        collection.InsertOne(document);
+                        break;
+
+                    case "replaceOne":
+                        var replaceOneFilter = Builders<BsonDocument>.Filter.Eq("_id", 1);
+                        var replacement = new BsonDocument("_id", 1);
+                        collection.ReplaceOne(replaceOneFilter, replacement);
+                        break;
+
+                    case "updateOne":
+                        var updateOneFilter = Builders<BsonDocument>.Filter.Eq("_id", 1);
+                        var updateOne = Builders<BsonDocument>.Update.Set("x", 2);
+                        collection.UpdateOne(updateOneFilter, updateOne);
+                        break;
+
+                    default:
+                        throw new Exception($"Unexpected operation: {operation}.");
+                }
+
+                AssertCommandHasTransactionId(eventCapturer);
+            }
+        }
+
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void Supported_multi_statement_writes_should_have_transaction_id(
```

Comment:
tests ordering

Reply:
Let's keep the spec order instead of alphabetically